src/Domain/Entities/TodoItem.cs

```diff
@@ -1,16 +1,54 @@
 ﻿using SSW.CleanArchitecture.Domain.Common.Base;
-using SSW.CleanArchitecture.Domain.Enum;
+using SSW.CleanArchitecture.Domain.Events;
 
 namespace SSW.CleanArchitecture.Domain.Entities;
 
-
 public record TodoItemId(Guid Value);
 
 public class TodoItem : BaseEntity<TodoItemId>
 {
-    public string? Title { get; set; }
+    // NOTE: private setters for behavior we want to encapsulate, and public setters for properties that don't have behavior
+
+    public string? Title { get; private set; }
     public string? Note { get; set; }
     public PriorityLevel Priority { get; set; }
     public DateTime Reminder { get; set; }
-    public bool Done { get; set; }
+    public bool Done { get; private set; }
+
+    private TodoItem() { }
+
+    public static TodoItem Create(string title)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(title, nameof(title));
+
+        var todoItem = new TodoItem
+        {
+            Id = new TodoItemId(Guid.NewGuid()),
+            Title = title,
+            Priority = PriorityLevel.None,
+            Done = false
+
+        };
+
+        todoItem.AddDomainEvent(new TodoItemCreatedEvent(todoItem));

```

Comment:
Needed to encapsulate creation of `TodoItem` so that we can ensure domain events are always fired.